Mathlib/Order/BourbakiWitt.lean

```diff
@@ -0,0 +1,227 @@
+(which we call `cSup`) -/
+class ChainCompletePartialOrder (α : Type) [Nonempty α] extends PartialOrder α where
+  /-- The supremum of a non-empty chain -/
+  cSup : NonemptyChain α → α
+  /-- `cSup` is an upper bound of the non-empty chain -/
+  le_cSup (c : NonemptyChain α) (x : α) : x ∈ c.carrier → x ≤ cSup c
+  /-- `cSup` is a lower bound of the set of upper bounds of the non-empty chain -/
+  cSup_le (c : NonemptyChain α) (x : α) : (∀ y ∈ c.carrier, y ≤ x) → cSup c ≤ x
+
+/-- A function `f : α → α` is inflationary if for all `x : α`, `x ≤ f x` -/
+structure Inflationary (α : Type) [PartialOrder α] where
+  /-- The underlying function of an inflationary function -/
+  toFun : α → α
+  /-- The underlying function of an inflationary function is inflationary -/
+  le_map (x : α) : x ≤ toFun x
+
+instance {α : Type} [PartialOrder α] : CoeFun (Inflationary α) (fun _ ↦ α → α) where
+  coe := Inflationary.toFun
+
+open ChainCompletePartialOrder Set
+
+section
+
+namespace BourbakiWitt
+
+variable {α : Type} [Nonempty α] [ChainCompletePartialOrder α] {x : α} {f : Inflationary α}
+
+/-- The definition of an admissible set with base point `x` and inflationary function `f` -/
+structure IsAdmissible (x : α) (f : Inflationary α) (s : Set α) : Prop where
+  /-- The base point is the least element of an admissible set -/
+  base_isLeast : IsLeast s x
+  /-- The image of an admissible set under the inflationary function is a subset of itself -/
+  image_self_sub_self : f '' s ⊆ s
+  /-- If a chain is a subset of an admissible set, its `cSup` is a member of the admissible set -/
+  cSup_mem : ∀ (c : NonemptyChain α), ↑c ⊆ s → cSup c ∈ s
+
+lemma ici_isAdmissible : IsAdmissible x f (Ici x) where
+  base_isLeast := ⟨le_refl x, fun _ h ↦ h⟩
+  image_self_sub_self := by
+    rintro _ ⟨y, hy, rfl⟩
+    exact le_trans hy (f.le_map _)
+  cSup_mem := by
+    intro c hc
+    have ⟨y, hy⟩ := c.Nonempty'
+    apply le_trans (hc hy) (le_cSup _ _ hy)
+
+/-- The bottom admissible set with base point `x` and inflationary function `f` -/
+abbrev bot (x : α) (f : Inflationary α) : Set α := ⋂₀ {s | IsAdmissible x f s}
+
+lemma mem_bot_iff {y : α} :
+    y ∈ bot x f ↔ ∀ s ∈ {s | IsAdmissible x f s}, y ∈ s := by
+  unfold bot
+  exact mem_sInter
+
+lemma bot_isAdmissible : IsAdmissible x f (bot x f) where
+  base_isLeast := by
+    constructor
+    · rw [mem_bot_iff]
+      exact fun _ h ↦ h.base_isLeast.1
+    · intro y hy
+      rw [mem_bot_iff] at hy
+      exact (hy (Ici x) ici_isAdmissible)
+  image_self_sub_self := by
+    rintro _ ⟨y, hy, rfl⟩
+    rw [mem_bot_iff]
+    rintro s hs
+    exact hs.image_self_sub_self ⟨y, ⟨mem_sInter.1 hy _ hs, rfl⟩⟩
+  cSup_mem := by
+    intro c hc
+    rw [mem_bot_iff]
+    intro s hs
+    exact hs.cSup_mem c (subset_trans hc (sInter_subset_of_mem hs))
+
+lemma sub_bot_eq_bot {s : Set α} (h : IsAdmissible x f s) (h' : s ⊆ bot x f) : s = bot x f :=
```

Comment:
`sub` is always an abbreviation for subtraction `(· - ·)`. To match the existing `le_bot_iff`, how about:

```suggestion
lemma subset_bot_iff {s : Set α} (h : IsAdmissible x f s) : s ⊆ bot x f ↔ s = bot x f :=
```